pkg/export/gcm/promtest/promtest.go

```diff
@@ -0,0 +1,316 @@
+package promtest
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/efficientgo/e2e"
+	"github.com/go-kit/log"
+	"github.com/gogo/protobuf/proto"
+	"github.com/google/go-cmp/cmp"
+	"github.com/oklog/ulid"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/model/timestamp"
+	"github.com/thanos-io/thanos/pkg/runutil"
+)
+
+// GCMServiceAccountOrFail gets the Google SA JSON content from GCM_SECRET
+// environment variable or fails.
+func GCMServiceAccountOrFail(t testing.TB) []byte {
+	// TODO(bwplotka): Move it to https://cloud.google.com/build CI.
+	saJSON := []byte(os.Getenv("GCM_SECRET"))
```

Comment:
Would it maybe be cleaner and more generic to use flags?

Reply:
Hm what's the benefit? What being generic here means/gives us? 

I made it an env var as flags in tests are very rare in OSS e.g. thanks of your comments I know it's possible (found [blog posts](https://blog.jbowen.dev/2019/08/using-go-flags-in-tests/) about it). 🙈 

I guess flags are easier to discover perhaps? But not sure how this applies here. Happy to learn about benefits of flags in tests here, so far it feels it does not matter and envvar is more consistent (we store secret in envvar too)

Reply:
Just thought it would be discoverable and cleaner to pass into the code, but you're also right that it doesn't bring much benefit. We use flags in our current `e2e` tests. Maybe that's more okay there because the flags are global. Thanks!